### ISSUE TITLE
Tier 2 test farm: setup an MVP against BitBar

### ISSUE THREAD

**Issue by rpappalax** [mozilla-mobile/mobile-test-eng#74]

**Comment by rpappalax** [comment]
notes from Johan, Andrew


- Is https://firefox-source-docs.mozilla.org/taskcluster/taskgraph.html
generally applicable to Fenix's taskgraph?

Yes.

- Is there a mach equivalent that lets users trigger jobs on
non-mozilla-central repos?
  - can run  on PR's.

put the following into the kind.yml, and then PR's pushes should create jobs.
  run-on-tasks-for: allows me to test while developing
    -     run-on-tasks-for: ["github-pull-request"]



////

fenix taskgraph overview:
https://johanlorenzo.github.io/blog/2019/10/24/taskgraph-is-now-deployed-to-the-biggest-mozilla-mobile-projects.html

- taskgraph labels are always unique

generally how to change taskgraph
  - create new task
     - this wouuld probably be a new kind.

you can modify the taskgraph in a variety of ways, but generally
best/Johan's favorite way:
 1. create a new kind
     2. configuration in the kind
     - transforms data to be more complete
     - kind for defining a different type


There's already a job that targets Bitbar infrastructure:
look at taskcluster/ci/browsertime/kind.yml (in fenix repo)


https://github.com/mozilla-mobile/fenix/blob/master/taskcluster/fenix_taskgraph/transforms/browsertime.py#L75


- produced artifacts: line 65
- dependencies: line 51


Just copy that file, add the run on pr bits, change the targeted queue
to test queue, create PR, and jobs should start hitting the correct
queue.

Talk to Johan and/or Tom Prince (in my TZ) for help.

--


Johan asks: On IOS they use Fastlane on Bitrise with emulators for
screenshots, why can't they use them for Android.
  - https://docs.fastlane.tools/actions/capture_ios_screenshots/#parameters
    - Plugin only available for IOS. Emulator must only provide it on IOS.

**Comment by rpappalax** [comment]
see: https://github.com/mozilla-mobile/android-components/issues/2331

**Comment by rpappalax** [comment]
waiting on aerickson to get first pass in place
